netstandard/ref/System.Runtime.CompilerServices.cs

```diff
@@ -131,13 +131,18 @@ public partial class CompilerGlobalScopeAttribute : System.Attribute
     public static partial class CompilerMarshalOverride
     {
     }
-    public sealed partial class ConditionalWeakTable<TKey, TValue> where TKey : class where TValue : class
+    public sealed partial class ConditionalWeakTable<TKey, TValue> : System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<TKey, TValue>>, System.Collections.IEnumerable where TKey : class where TValue : class
     {
         public ConditionalWeakTable() { }
         public void Add(TKey key, TValue value) { }
+        public void AddOrUpdate(TKey key, TValue value) { }
+        public void Clear() { }
+        ~ConditionalWeakTable() { }
```

Comment:
If we stick with our other decision we should exclude destructors on sealed types.

Reply:
Does this have any impact from the compiler side of things? CC. @jaredpar

Reply:
Talked with @jaredpar briefly at lunch.... His comment was essentially: If you don't have a good reason to exclude it, don't.

Also, something about potentially breaking or hindering certain analyzers that provide diagnostics based on the existence of a finalizer.

Reply:
ConditionalWeakTable does not have finalizer in the implementation:

https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

It would be remove it the .NET Core ref assembly instead of adding it here.

Reply:
> ConditionalWeakTable does not have finalizer in the implementation

Seems like a bug in the reference assembly generation then?

Reply:
The reference assembly checker allows for harmless differences. ConditionalWeakTable had finalizer in desktop originally. The finalizer was removed as part of .NET Core optimizations. Adding/removing finalizers on sealed types is non-breaking change and thus not flagged as a problem y the reference assembly checker.

Reply:
Alright, I've removed the finalizer here.